CRM/Utils/Recent.php

```diff
@@ -188,4 +204,51 @@ public static function delContact($id) {
     $session->set(self::STORE_NAME, self::$_recent);
   }
 
+  /**
+   * Check if a provider is allowed to add stuff.
+   * If correspondig setting is empty, all are allowed
+   *
+   * @param string $providerName
+   */
+  public static function isProviderEnabled($providerName) {
```

Comment:
Do you think this method is necessary to be public , Can you see it can be useful and may be used outside the context of this class by someone else ... If not then please make it private I don't see a point for it to be public .